lib/src/server/handler.dart

```diff
@@ -313,11 +313,17 @@ class ServerHandler extends ServiceCall {
       _stream.sendData(frame(bytes, _callEncodingCodec));
     } catch (error, trace) {
       final grpcError = GrpcError.internal('Error sending response: $error');
-      if (!_requests!.isClosed) {
-        // If we can, alert the handler that things are going wrong.
-        _requests!
-          ..addError(grpcError)
-          ..close();
+      // Safely attempt to notify the handler about the error
+      // Use try-catch to prevent "Cannot add event after closing" from crashing the server
+      if (_requests != null && !_requests!.isClosed) {
+        try {
+          _requests!
+            ..addError(grpcError)
+            ..close();
+        } catch (e) {
+          // Stream was closed between check and add - ignore this error
+          // The handler has already been notified or terminated
+        }
```

Comment:
do we want to log to sentry in the case there is an error here? or just ignore? or potentially some log in the terminal?